test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java

```diff
@@ -223,6 +221,8 @@
     private ServiceDisruptionScheme activeDisruptionScheme;
     private Function<Client, Client> clientWrapper;
 
+    private final Object discoveryFileMutex = new Object();
```

Comment:
I would prefer to just move this next to the `rebuildUnicastHostFiles` method as it's only used there.